src/sandstorm/sandstorm-http-bridge.c++

```diff
@@ -1238,7 +1290,19 @@ private:
     auto additionalHeaderList = context.getAdditionalHeaders();
     if (additionalHeaderList.size() > 0) {
       for (auto header : additionalHeaderList) {
-        lines.add(kj::str(header.getName(), ": ", header.getValue()));
+        if (checkIsHttpToken(header.getName()) && checkIsHeaderValue(header.getValue())) {
+          // Check header name and value to prevent header injection.
+          if (header.getPrefixed()) {
+            // If the header has the "prefixed" flag set, prepend the app header prefix.
+            lines.add(kj::str(
+                  WebSession::APP_HEADER_PREFIX.toString(),
+                  header.getName(),
+                  ": ",
+                  header.getValue()));
+          } else {
+            lines.add(kj::str(header.getName(), ": ", header.getValue()));
```

Comment:
Can you add a check that the header is actually in the whitelist here?

Note that since the whitelist isn't long, it's probably fine to iterate over it.